Source/Shaders/GlobeVS.glsl

```diff
@@ -167,4 +172,11 @@ void main()
     v_rayleighColor = atmosColor.rayleigh;
     v_distance = length((czm_modelView3D * vec4(position3DWC, 1.0)).xyz);
 #endif
+
+#ifdef APPLY_MATERIAL
+    vec3 finalNormal = normalize(v_normalMC);
+    vec3 worldNormal = normalize(v_positionMC.xyz);
```

Comment:
Here and the line above, we generally try to treat varyings in a vertex shader as "write once, don't read" - more for style - I don't think it matters for performance.

Here, the return from `czm_octDecode` could be assigned to a local, and `position3DWC` can be used instead of `v_positionMC`.

Reply:
Also `worldNormal` is a confusing name - since it is not the terrain normal in world coordinate - perhaps rename to `ellipsoidNormal` since this is normal to the ellipsoid?